akka-docs/src/main/paradox/typed/persistence-durable-state.md

```diff
@@ -0,0 +1,391 @@
+* `IncrementBy` increments the counter by the value passed to it and persists the updated value as an effect in the State
+* `GetValue` retrieves the value of the counter from the State and replies with it to the actor passed in
+
+Scala
+:  @@snip [DurableStatePersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/DurableStatePersistentBehaviorCompileOnly.scala) { #command-handler }
+
+Java
+:  @@snip [DurableStatePersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java) { #command-handler }
+
+@scala[These are used to create a `DurableStateBehavior`:]
+@java[These are defined in an `DurableStateBehavior`:]
+
+Scala
+:  @@snip [DurableStatePersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/DurableStatePersistentBehaviorCompileOnly.scala) { #behavior }
+
+Java
+:  @@snip [DurableStatePersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java) { #behavior }
+
+## Effects and Side Effects
+
+A command handler returns an `Effect` directive that defines what state, if any, to persist. 
+Effects are created using @java[a factory that is returned via the `Effect()` method] @scala[the `Effect` factory]
+and can be one of: 
+
+* `persist` will persist state atomically, i.e. all states are stored or none of them are stored if there is an error
+* `none` no state to be persisted, for example a read-only command
+* `unhandled` the command is unhandled (not supported) in current state
+* `stop` stop this actor
+* `stash` the current command is stashed
+* `unstashAll` process the commands that were stashed with @scala[`Effect.stash`]@java[`Effect().stash`]
+* `reply` send a reply message to the given `ActorRef`
+
+Note that only one of those can be chosen per incoming command. It is not possible to both persist and say none/unhandled.
+
+In addition to returning the primary `Effect` for the command `DurableStateBehavior`s can also 
+chain side effects that are to be performed after successful persist which is achieved with the `thenRun`
+function that runs the callback passed to it e.g @scala[`Effect.persist(..).thenRun`]@java[`Effect().persist(..).thenRun`]. 
+
+All `thenRun` registered callbacks are executed sequentially after successful execution of the persist statement
+(or immediately, in case of `none` and `unhandled`).
+
+In addition to `thenRun` the following actions can also be performed after successful persist:
+
+* `thenStop` the actor will be stopped
+* `thenUnstashAll` process the commands that were stashed with @scala[`Effect.stash`]@java[`Effect().stash`]
+* `thenReply` send a reply message to the given `ActorRef`
+
+In the example below, we use a different constructor of `DurableStateBehavior.withEnforcedReplies`, which creates
```

Comment:
What does it mean that the reply is not forgotten? why would a reply be forgotten without this? (trying to read this as a new reader, it's not very clear to me, it assumes some knowledge of what can go wrong if I don't use it.)

Reply:
Oh, maybe this means that you could forget to define a reply, as the person writing this, and by using this constructor, you can't define a command that does not have a reply. 

You could misread this as if Akka forgets replies. So maybe make it clear that this is related to the case where you as a developer want to ensure this.

Reply:
refactored this part quite a bit ..